### ISSUE TITLE
Struggling with hot reloading config

### ISSUE THREAD

**Issue by nick13jaremek** [gaearon/react-hot-loader#988]
### Description

Followed the `react-hot-loader` [setup instructions for a just-ejected `create-react-app` application](https://github.com/gaearon/react-hot-loader#migrating-from-create-react-app). Hot reloading is not working on child components of main `App` component: a `Button` component has its label's font weight set to *bold* but hot reloading does not produce this change on the browser.

However, applying changes to the main `App` component **does trigger** hot reloading.

### Expected behavior

Whenever a child component of the main `App` component has its logic updated, the app running in the browser should hot reload, mantaining the current components' state.

### Actual behavior

Changing the logic of a main `App` component's child does not trigger hot reloading, keeping the child component unchanged.

### Environment

React Hot Loader version: 4.2.0

Run these commands in the project folder and fill in their results:

1.  `node -v`: 6.9.5
2.  `npm -v`: 3.10.10

Then, specify:

1.  Operating system: Ubuntu 16.04
2.  Browser and version: Google Chrome | 63.0.3239.108

### Reproducible Demo

https://github.com/nick13jaremek/my-react-app

1. Clone the repo.
2. Run `npm install`.
3. Run `npm start`. A browser tab should open with the app running.
4. Click on the shown button to increase the click counter. 
5. Open `src/Button.js` and change the `style` variable to contain the following value: `{ fontWeight: 'bold' }`.
6. The desired output should yield a button with a bolded text, but keeping the click counter value, due to hot reloading. However, **this does not happen**.

Any help with this issue would be greatly appreciated.

**Comment by theKashey** [comment]
Yeah, sorry but right now RHL does not update objects, like state or .... style. Nor objects, nor arrays - only simple fields and functions.
This is tracked by #840 and planned to be fixed in a short terms.
it is just not super _safe_ to _fix_ it, as long it could break way more cases than fix.

**Comment by nick13jaremek** [comment]
@theKashey Thank you for the clarification.

I was following instructions from a book to enable hot-reloading on the styling of a `Button` component, as outlined in my previous comment.

However, due to a personal typo, I noticed that the `style` variable in the `Button` component **must be outside** the `Button` class definition. Then, changing this `style` variable would update the component styling, while persisting its state. I achieved so after fixing the typo. 

My apologies for not noticing the typo earlier. Now it all makes sense based on your response.
